lib/CompilerImpl/ClangLLVM/LLVMInstanceManager.cpp

```diff
@@ -97,9 +97,14 @@ void LLVMInstanceManager::initializeLLVM() {
   llvm::initializeTarget(passRegistry);
   // For codegen passes, only passes that do IR to IR transformation are
   // supported.
-  llvm::initializeCodeGenPreparePass(passRegistry);
+#if LLVM_VERSION_MAJOR > 18
+  llvm::initializeAtomicExpandLegacyPass(passRegistry);
+  llvm::initializeCodeGenPrepareLegacyPassPass(passRegistry);
+#else
   llvm::initializeAtomicExpandPass(passRegistry);
-  llvm::initializeRewriteSymbolsLegacyPassPass(passRegistry);
+  llvm::initializeCodeGenPreparePass(passRegistry);
+  llvm::initializeRewriteSymbols(passRegistry); // not sure about LLVM 16 or 17, not available in LLVM versions 18/19
```

Comment:
can you try it with the latest supported version of LLVM that libVC has?
We can always declare unsupported too old versions.